netutils/netinit/netinit.c

```diff
@@ -357,6 +365,85 @@ static void netinit_set_macaddr(void)
 #  define netinit_set_macaddr()
 #endif
 
+#ifdef CONFIG_BOARDCTL_NETCONF
+/****************************************************************************
+ * Name: netinit_get_ipaddrs
+ *
+ * Description:
+ *   Setup IP addresses.
+ *
+ *   For 6LoWPAN, the IP address derives from the MAC address.  Setting it
+ *   to any user provided value is asking for trouble.
+ *
+ ****************************************************************************/
+
+static int netinit_get_ipaddrs(void)
+{
+#ifdef CONFIG_NETINIT_DHCPC
+  bool use_dhcp   = false;
+#endif
+  bool use_static = false;
+#ifdef CONFIG_NET_IPv4
+  struct in_addr addr;
+
+  int ret = boardctl(BOARDIOC_NETCONF, (uintptr_t) &g_netconf);
```

Comment:
why not support ipv6

Reply:
As I disucss in apache/incubator-nuttx#1886, I do not think it is appropriate to use boardctl() for this purpose when the established, mostly standard ways of getting this information.  NuttX is ALL about conformance to standards and we must be faithful to that.

boardctl() is unfortunately necessary, but we cannot let it be an expedient, low-effort solution for system interface issues that already have standard solutions.  We must to the work as necessary and not cut corners.